queues/queue.go

```diff
@@ -0,0 +1,34 @@
+package queues
+
+import "github.com/akitasoftware/go-utils/optionals"
+
+type Queue[T any] struct {
```

Comment:
I think a comment that it's a FIFO queue would be appropriate.

Reply:
Ah, versus a priority queue. In 446e7a6.